### ISSUE TITLE
Collapse plugin improved accessibility using aria roles and states / keyboard listeners 

### ISSUE THREAD

**Issue by bassettsj** [twbs/bootstrap#13555]
Again from the [paypal/bootstrap-accessibility-plugin](https://github.com/paypal/bootstrap-accessibility-plugin/#collapse), these changes should be added to the documentation and the JS plugin.

### Collapse 
1. Add ~~tab role~~, aria-selected, ~~aria-expanded~~, ~~aria-controls~~, and tabIndex for collapse tab.
2. Add ARIA roles of ~~tabPanel~~, tabIndex, ~~aria-hidden~~, and ~~aria-labelledBy~~ for collapsible panel.
3. ~~Add role of tabList and aria-multiselectable for collapse container div.~~
4. Dynamically flip tabIndex, aria-selected, and ~~aria-expanded~~ for tab when it is activated and add aria-hidden to hide the previously visible collapse tabpanel.
5. Add keydown event listener for the collapse component to work with keyboard.

**Comment by mdo** [comment]
What's the status of this?

/cc @patrickhlauke

**Comment by patrickhlauke** [comment]
I believe that the approach taken by the PayPal plugin is a tad excessive - treating a simple disclosure widget as a tab panel is, in my view, overcomplicating things. As such I'd say that any outstanding items are unnecessary.

**Comment by patrickhlauke** [comment]
Sorry, to clarify the comment above: the "excessive" part refers to the simple case of a button/link that shows/hides a single panel - a classic disclosure widget.
Once you move into "accordion" territory, the PayPal approach is fair...but we'll revisit this together with the more traditional tab panel fixes from https://github.com/twbs/bootstrap/issues/16753

**Comment by RichCaloggero** [comment]
I just downloaded bs 3.7 and simple disclosure widgit does not toggle `aria-expanded`.

Hopefully, someone can include this fix for the simple use case of a button controlling a collapsable.

**Comment by cvrebert** [comment]
(a) I am unable to reproduce your problem.
(b) Unfortunately, v3 is no longer supported; see #20631.
